src/crimson/net/Protocol.cc

```diff
@@ -55,10 +55,11 @@ seastar::future<> Protocol::close()
   assert(!close_ready.valid());
 
   if (socket) {
-    close_ready = socket->close()
-      .then([this] {
-        return pending_dispatch.close();
-      }).finally(std::move(cleanup));
+    socket->shutdown();
+    close_ready = pending_dispatch.close(
+    ).finally([this] {
```

Comment:
nit, please move this line up to the previous one.

Reply:
done.